packages/legend-application/src/components/shared/BasicValueSpecificationEditor.tsx

```diff
@@ -604,6 +684,16 @@ export const BasicValueSpecificationEditor: React.FC<{
   className?: string | undefined;
   setValueSpecification: (val: ValueSpecification) => void;
   resetValue: () => void;
+  selectorConfig?:
```

Comment:
I'm not entirely convinced of this approach, I think what we should do is just let people pass in a function to return `Promise<string[]>` and we keep all the other states (loading, results) as part of this component or so.